### ISSUE TITLE
Masked dotplot looks for genes in adata.raw

### ISSUE THREAD

**Issue by simomounir** [theislab/spapros#257]
**Description**

I am trying to use this plotting tool provided with spapros in order to summarise some example adata.

sp.pl.masked_dotplot(adata, selector, save=True)

I attached a text file of the error:

[masked_dotplot_bug.txt](https://github.com/theislab/spapros/files/11062798/masked_dotplot_bug.txt)

The problem seems to be that the plotting function used requires the gene ids to be in adata.raw.var_names when I know I have them in adata.var_names. IT must be some parameter that can easily force the plotting functions within the spapros customized one to use the names from adata.var_names (use_raw=False?).

Or can you suggest some other idea on how to circumvent this bug?

This issue seems to be common when people load their data from an "h5ad" object.

**To Reproduce**

1. Loaded file from with the following:

adata = sc.read_h5ad("data.h5ad")

2. I run the selector function to get marker genes

selector.select_probeset()

3.  run the following plotting function:

sp.pl.masked_dotplot(adata, selector, save=True)

**Expected behavior**

**System [Windows 10]:**

-   Language Version: [e.g. Python 3.9.0]
-   Virtual environment: [e.g. Conda]

**Additional context**

**Comment by LouisK92** [comment]
I see: If `raw` is present scanpy's plotting functions look there by default. Unfortunately the `sp.pl.masked_dotplot` currently doesn't have enabled the `use_raw` argument or in general `**kwargs` for scanpy's dotplot. That's something that 
- [ ]  should be added.
 
Currently the issue should be solved with `del adata.raw` - would that work for you?

**Comment by simomounir** [comment]
Tried it but I get some other error:

[masked_dotplot_bug.txt](https://github.com/theislab/spapros/files/11063367/masked_dotplot_bug.txt)

**Comment by LouisK92** [comment]
Ah `save` expects either `False` or a `str` (i.e. the path where the img is saved e.g. `./masked_dotplot.png`)

**Comment by simomounir** [comment]
That fixed it. Thanks a lot :)

**Comment by LouisK92** [comment]
This is now solved with commit ddd5cb7ddc72244eff7ee8e032501314b42fd475.

The keyword `use_raw` was added and is set to `False` by default.

Quick example to show the behaviour:

```python
import spapros as sp
adata = sp.ut.get_processed_pbmc_data()
selector = sp.se.ProbesetSelector(
    adata, n_pca_genes=0, n=20, celltype_key="celltype", verbosity=1, save_dir=False
)
selector.select_probeset()

adata2 = adata.copy()
adata2.X = adata2.X.toarray()
adata2.raw = adata2.copy()
adata2.X += 1

sp.pl.masked_dotplot(adata2,selector) # Doesn't use raw counts --> expect full circles (due to +1)
sp.pl.masked_dotplot(adata2,selector,use_raw=True) # Uses raw counts --> expect unchanged circles
```
